### DISCUSSION TITLE
How to mesh a closed body with unregular shape?

### DISCUSSION THREAD

**Discussion by fengzekang** [neperfepx/neper#824] [Q&A] [answered]
Hello developer and all:
   If I want to mesh the voronoi meshes for a closed body with unregular shape, what should I do?
   I noticed that we can mesh a cube(such as 1:1:1), and also we can define the mesh parameters to control the quality. But if I just want to mesh a 3D closed circular ring body, what should I do. I mean is there any example or the input files and commands?
   Thanks a lot. Best regards.
Feng

**Comment by rquey** [comment, marked as answer]
Good idea for a future tutorial.  Meanwhile, you can check out [here](https://neper.info/generation.html#generalities) for a few examples. You'd have to use [`-domain`](https://neper.info/doc/neper_t.html#cmdoption-domain) and optionally [`-transform cut`](https://neper.info/doc/neper_t.html#cmdoption-transform) to create non-convex shapes.
